### ISSUE TITLE
Password validation is failing

### ISSUE THREAD

**Issue by drujensen** [amberframework/amber#1174]
### Description

Password validation is failing when trying to login.

### Steps to Reproduce

```sh
amber new blog
cd blog
amber g auth user
amber db create migrate
amber watch
```

1. Access http://localhost:3000
2. Create a new account test@test.com / password
3. Logout
4. Login with test@test.com / password

**Expected behavior:** [What you expect to happen]

User is logged in

**Actual behavior:** [What actually happens]

Error saying email or password is incorrect

**Reproduces how often:** [What percentage of the time does it reproduce?]

100%

### Versions

crystal 0.32.0 / amber 0.32.0

**Comment by Brodan** [comment]
I'm also seeing this currently. When I check the DB I see the user created and the `hashed_password` column is completely blank, thus re-logging in fails. This is when using postgres if that helps.

**Comment by andrewc910** [comment]
It looks like `User#password=` is not being invoked with `user = User.new user_params.validate!`(user_controller#create). Actually, it looks like no manual setters are invoked like this. 

Adding `user_params.validate!["password"]` below the initialization of the `User` object ensures `User#password=` is invoked. 

I believe this fix will take a deep dive into Granite or Amber Router. I implemented a basic `email=` to see if it would be invoked and it is not. I tried something like the following which failed to hit:
```crystal
def email=(email)
  @email = email
end
```

And i tried(just to see logs) which again failed:
```crystal
def email=(email)
  pp "EMAIL INVOKED"
  pp "EMAIL INVOKED"
  pp "EMAIL INVOKED"
  pp "EMAIL INVOKED"
  @email = email
end
```

It looks like either the router or granite is bypassing these setters(some how). I will open a PR with quick fix i provided above but i do believe their is some bigger underlying issue.
